### DISCUSSION TITLE
Can't ignore php update

### DISCUSSION THREAD

**Discussion by pimjansen** [renovatebot/renovate#18905] [Request Help] [answered]
Hey,

Im starting to integrate Renovate which goes fine. I get 2 PRs one for my PHP deps and one for my NPM deps. However a standalone PR is generate for the PHP update itself which i would love to ignore in general. 

In general i also have no idea where he is getting it from since it is not a packagist data reference if im correct?

Any ideas?
Thanks!

## Current output
```
php | require | patch | ^8.1 -> ^8.1.12
```

## Config

```json
{
  "platform": "azure",
  "endpoint": "https://org.visualstudio.com/",
  "repositories": [
    "Infrastructure/sla"
  ],
  "extends": [
    ":dependencyDashboard",
    ":ignoreModulesAndTests",
    ":rebaseStalePrs",
    ":semanticCommitsDisabled",
    ":separateMajorReleases",
    ":combinePatchMinorReleases",
    ":enableVulnerabilityAlerts",
    ":label(Renovate)",
    ":gitSignOff"
  ],
  "vulnerabilityAlerts": {
    "extends": [
      ":labels(Security CVE)"
    ],
    "rangeStrategy": "bump"
  },
  "labels": ["dependencies"],
  "lockFileMaintenance": {
    "enabled": true
  },
  "rollbackPrs": true,
  "ignoreUnstable": true,
  "recreateClosed": true,
  "rangeStrategy": "bump",
  "rebaseWhen": "auto",
  "packageRules": [{
    "matchDatasources": ["packagist"],
    "groupName": "PHP packages",
    "groupSlug": "PHP"
  }, {
    "matchDatasources": ["npm"],
    "groupName": "NPM packages",
    "groupSlug": "NPM",
    "stabilityDays": 3
  }],
  "major": {
    "enabled": false
  }
}
```

**Comment by pimjansen** [comment]
This also will not do the trick btw:

```json
  "packageRules": [{
    "matchManagers": ["composer"],
    "excludePackageNames": ["php"]
  },{
    "matchDatasources": ["packagist"],
    "groupName": "PHP packages",
    "groupSlug": "PHP"
  }, {
    "matchDatasources": ["npm"],
    "groupName": "NPM packages",
    "groupSlug": "NPM",
    "stabilityDays": 3
  }],
```

**Reply by PhilipAbed** [reply]
```
  "packageRules": [
    {
      "matchPackagePatterns": ["^php"],
      "excludePackageNames": ["php"]
    }
  ]
```  
and put it as the last PackageRule, order of package rules matter
if that didn't work i would like to see the PR or a reproduction repo

**Reply by rarkins** [reply, marked as answer]
This rule achieves nothing, no matter where you put it. `match*` and `excluse*` rules are just used for matching, and do nothing unless there's also an action to do. Therefore you need this, and it can go anywhere in the packageRules:

```
    {
      "matchManagers": ["composer"],
      "matchPackageNames": ["php"],
      "enabled": false
    }
```

**Reply by pimjansen** [reply]
@rarkins thanks! I was indeed looking into @PhilipAbed way since that feels a bit more natural to start with but understand your answer as well. Selecting and ignoring at the same time also does not feel right.

**Reply by PhilipAbed** [reply]
https://docs.renovatebot.com/configuration-options/#excludepackagenames

i actually thought the `excludePackageNames` is an action.

it says: `Use excludePackageNames if you want to have one or more exact name matches excluded in your package rule` though, but doesn't give an example on how to disable the package updates, 
if it had `enabled:false` it would've been more understandable

**Reply by PhilipAbed** [reply]
@pimjansen my rule won't work
it's actually excluding the match, not disabling the package.

you should match php pattern or name, then enable=false as rarkins suggested,
do not use the Exclude, as it's part of the match exclusions and not an action

**Reply by dbu** [reply]
i have a project on PHP 8.2 that i currently can't upgrade to 8.3

when i try [rarkin's answer](https://github.com/renovatebot/renovate/discussions/18905#discussioncomment-4134269), i get an error report about "Artifact update problem" with 

```
Command failed: composer update nelmio/api-doc-bundle:4.30.0 php:8.3.11 phpunit/phpunit:10.5.34 --with-dependencies ...
...

In RuleSetGenerator.php line 260:
                                                          
  [LogicException]                                        
  Fixed package php 8.2.23 was not added to solver pool. 
```

how do i correctly lock the php version for renovate?
